apple/swift-docc
# Fix encoding & decoding inconsistencies for link destination summary variant values

Bug/issue #, if applicable: rdar://103916556

## Summary

This fixes two bugs in the encoding and decoding of variant values for link destination summaries. 

## Dependencies

n/a

## Testing

It's a bit cumbersome to test this end-to-end to verify that DocC can correctly decode the linkable entities that it encodes. The updated unit tests cover this. 

If we wanted to test it manually we would need to:
- Build multi-source-language documentation and export the linkable entities.
- Create a custom external resolver and use it to return the data from the first build
- Build other documentation which uses the above custom external resolver to return the data from the first build.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Updated documentation if necessary (if OpenAPI spec is considered documentation)
